src/warnet/k8s.py

```diff
@@ -104,29 +141,54 @@ def apply_kubernetes_yaml_obj(yaml_obj: str) -> None:
         Path(temp_file_path).unlink()
 
 
-def delete_namespace(namespace: str) -> bool:
-    command = f"kubectl delete namespace {namespace} --ignore-not-found"
-    return run_command(command)
+def delete_namespace(namespace: str) -> V1Status:
+    v1: CoreV1Api = get_static_client()
+    resp = v1.delete_namespace(namespace)
+    return resp
+
+
+def delete_pod(
+    pod_name: str,
+    namespace: str,
+    grace_period: int = 30,
+    force: bool = False,
+    ignore_not_found: bool = True,
+) -> Optional[V1Status]:
+    v1: CoreV1Api = get_static_client()
+    delete_options = V1DeleteOptions(
+        grace_period_seconds=grace_period,
+        propagation_policy="Foreground" if force else "Background",
+    )
+    try:
+        resp = v1.delete_namespaced_pod(name=pod_name, namespace=namespace, body=delete_options)
+        return resp
+    except ApiException as e:
+        if e.status == 404 and ignore_not_found:
+            print(f"Pod {pod_name} in namespace {namespace} not found, but ignoring as requested.")
+            return None
+        else:
+            raise
 
 
-def delete_pod(pod_name: str) -> bool:
-    command = f"kubectl delete pod {pod_name}"
-    return stream_command(command)
+def get_default_namespace() -> str:
+    with open(KUBECONFIG) as file:
+        kubeconfig_data = yaml.safe_load(file)
```

Comment:
will this file always exist? is this the only way to get the namespace?

Reply:
I will need to revisit this.

Reply:
I replaced open(KUBECONFIG) instances with two functions that handle errors: https://github.com/bitcoin-dev-project/warnet/pull/614/commits/b1365d3160ec51925923c6b0173ff502075d40d1

Regarding getting the namespace, I do believe that the namespace stored in the KUBECONFIG file is the source of truth for the namespace for user OR our default namespace which is "warnet" right now. We talked about changing that on the call to "default". I